src/Smd/Service.php

```diff
@@ -26,9 +44,9 @@ class Service
      * @var string
      */
     protected $envelope  = Smd::ENV_JSONRPC_1;
-    protected $name;
+    protected $name = '';
```

Comment:
Not sure if I like this change. As I mentioned before - with PHP 7.4 it will be a bit easier. I think `null` for no initialised is fine now (looking at line 142)

Reply:
Removing empty string initialization means, that the return type of the getter is nullable. Would that not be confusing?

Also the [specs](https://github.com/dojo/docs/blob/master/dojox/rpc/smd.rst) state for `name`
> This should be unique for each method

From my understanding a nullable type is optional **and** not unique. Personally, I would prefer an empty string initialization.

Agreed on line 142, this should be a strict comparison.

Reply:
Ok, I think it's fine, because we have check in the constructor and providing empty name should be also invalid.